substrate/frame/scheduler/src/lib.rs

```diff
@@ -440,6 +462,86 @@ pub mod pallet {
 			)?;
 			Ok(())
 		}
+
+		/// Set a retry configuration for a task so that, in case its scheduled run fails, it will
+		/// be retried after `period` blocks, for a total amount of `retries` retries or until it
+		/// succeeds.
+		///
+		/// If a retried task runs successfully before running out of retries, its remaining retry
+		/// counter will be reset to the initial value. If a retried task runs out of retries, it
+		/// will be removed from the schedule.
+		///
+		/// Tasks which need to be scheduled for a retry are still subject to weight metering and
+		/// agenda space, same as a regular task. Periodic tasks will have their periodic schedule
+		/// put on hold while the task is retrying.
+		#[pallet::call_index(6)]
+		#[pallet::weight(<T as Config>::WeightInfo::set_retry(T::MaxScheduledPerBlock::get()))]
+		pub fn set_retry(
+			origin: OriginFor<T>,
+			task: TaskAddress<BlockNumberFor<T>>,
+			retries: u8,
+			period: BlockNumberFor<T>,
+		) -> DispatchResult {
+			T::ScheduleOrigin::ensure_origin(origin.clone())?;
+			let origin = <T as Config>::RuntimeOrigin::from(origin);
+			let (when, index) = task;
+			let agenda = Agenda::<T>::get(when);
+			match agenda.get(index as usize) {
+				Some(Some(task)) =>
+					if matches!(
+						T::OriginPrivilegeCmp::cmp_privilege(origin.caller(), &task.origin),
+						Some(Ordering::Less) | None
+					) {
```

Comment:
I see this copy-pasted a few times instead of being an `ensure_privilege` function, but does not need to be fixed in this MR.

Reply:
Moved it to `ensure_privilege`.